src/lightning_app/cli/cmd_pl_init.py

```diff
@@ -148,8 +149,17 @@ def print_pretty_report(
         text_pathname.stylize(f"link file://{path}")
         text_pathname.append(f" {padding} {help_text}", "blue")
 
-        icon = "📂 " if path.is_dir() else "📄 "
-        tree.add(Text(icon) + text_pathname)
+        icon = ""
+        try:
+            if path.is_dir():
+                icon = "📂 ".encode()
+            else:
+                icon = "📄 ".encode()
```

Comment:
Instead of encoding and decoding it again, let's just test whether it can be encoded or not. 
See the implementation here: https://github.com/Lightning-AI/lightning/blob/2622989b108368813fdd3850b002fd1ad69988c2/src/pytorch_lightning/loops/dataloader/evaluation_loop.py#L379-L385

You can create a small utility function:

```py
_can_encode_icon(icon: str) -> bool:
    try:
        icon.encode(sys.stdout.encoding)
        return True
    except UnicodeEncodeError:
        return False
```

This would then be easier to test as well.

Reply:
I have implemented the utility function and changed the encoding to include `sys.stdout.encoding`. Could you explain how to set up the terminal to display ascii for the purpose of testing? I am unable to replicate the bug.

Reply:
Great! Thanks. 
The issue I reported didn't occur to me directly, I just reported it from an internal user testing. I believe it was @carmocca and the terminal was the standard Ubuntu terminal.

Reply:
https://github.com/Lightning-AI/lightning/issues/12823 was the reported issue and #12828 has the fix proposed here with its test. In case you want inspiration

Reply:
I think the test already does the same now. Testing against anything that is not utf works I guess

Reply:
Is there anything else I should change to this? If not, what are the next steps I need to do?

Reply:
I think the issue is fixed, great work. The next step is review: At the bottom of the page you see that code-owner review is required from one of the maintainers listed there. After the missing approval is given, the PR will be merged. 
Thanks for your patience!